replay/models/cql.py

```diff
@@ -0,0 +1,357 @@
+            encoder factory for the critic.
+        q_func_factory (d3rlpy.models.q_functions.QFunctionFactory or str):
+            Q function factory.
+        batch_size (int): mini-batch size.
+        n_frames (int): the number of frames to stack for image observation.
+        n_steps (int): N-step TD calculation.
+        gamma (float): discount factor.
+        tau (float): target network synchronization coefficiency.
+        n_critics (int): the number of Q functions for ensemble.
+        initial_temperature (float): initial temperature value.
+        initial_alpha (float): initial :math:`\alpha` value.
+        alpha_threshold (float): threshold value described as :math:`\tau`.
+        conservative_weight (float): constant weight to scale conservative loss.
+        n_action_samples (int): the number of sampled actions to compute
+            :math:`\log{\sum_a \exp{Q(s, a)}}`.
+        soft_q_backup (bool): flag to use SAC-style backup.
+        use_gpu (bool, int or d3rlpy.gpu.Device):
+            flag to use GPU, device ID or device.
+        scaler (d3rlpy.preprocessing.Scaler or str): preprocessor.
+            The available options are `['pixel', 'min_max', 'standard']`.
+        action_scaler (d3rlpy.preprocessing.ActionScaler or str):
+            action preprocessor. The available options are ``['min_max']``.
+        reward_scaler (d3rlpy.preprocessing.RewardScaler or str):
+            reward preprocessor. The available options are
+            ``['clip', 'min_max', 'standard']``.
+        impl (d3rlpy.algos.torch.cql_impl.CQLImpl): algorithm implementation.
+
+    """
+
+    top_k: int
+    action_randomization_scale: float
+    n_epochs: int
+    model: CQL_d3rlpy.CQL
+
+    _search_space = {
+        "actor_learning_rate": {"type": "loguniform", "args": [1e-5, 1e-3]},
+        "critic_learning_rate": {"type": "loguniform", "args": [3e-5, 3e-4]},
+        "n_epochs": {"type": "int", "args": [3, 20]},
+        "action_randomization_scale": {"type": "loguniform", "args": [1e-3, 1e-1]},
+        "temp_learning_rate": {"type": "loguniform", "args": [1e-5, 1e-3]},
+        "alpha_learning_rate": {"type": "loguniform", "args": [1e-5, 1e-3]},
+        "gamma": {"type": "loguniform", "args": [0.9, 0.999]},
+        "n_critics": {"type": "int", "args": [2, 4]},
+    }
+    
+    def __init__(
+        self, *,
+        top_k: int, n_epochs: int = 1,
+        action_randomization_scale: float = 1e-3,
+
+        # CQL inner params
+        actor_learning_rate: float = 1e-4,
+        critic_learning_rate: float = 3e-4,
+        temp_learning_rate: float = 1e-4,
+        alpha_learning_rate: float = 1e-4,
+        actor_optim_factory: OptimizerFactory = AdamFactory(),
+        critic_optim_factory: OptimizerFactory = AdamFactory(),
+        temp_optim_factory: OptimizerFactory = AdamFactory(),
+        alpha_optim_factory: OptimizerFactory = AdamFactory(),
+        actor_encoder_factory: EncoderArg = "default",
```

Comment:
It should to describe allowed values for str params like  actor_encoder_factory,q_func_factory and so on

Reply:
Added to the class init args description corresponding lists of available options